### ISSUE TITLE
VSCE: Remove extension tests from main branch client checks

### ISSUE THREAD

**Issue by tjkandala** [sourcegraph/sourcegraph#34135]
We currently run integration tests for our VS Code and browser extensions on each main branch build. However, I do not believe they are necessary to ensure reliability of the extensions. We should stop running them on main branch client checks because:
- They have nothing to do with Sourcegraph "core" release process
- Our goal is to **not** have flakes, but we would have higher flake tolerance on, say, the browser extension release branch that we babysit when releasing the extension vs. on `main` where other engineers would be blocked by a flake

TODO:

- [x] Remove VS Code extension smoke test (#34136)
- [ ] Ensure extension tests run only on client PRs, nightly checks, and release branch
  - [ ] For browser extension
  - [ ] For VS Code extension
- [ ] Don't run web tests on `client/vscode` only changes (see https://github.com/sourcegraph/sourcegraph/pull/34213#issuecomment-1104231975). Tag FE platform to make sure there aren't any unforeseen consequences. Possibly w/ node modules?

**Comment by sourcegraph-bot-2** [comment]
Heads up @muratsu @jjinnii @ryankscott - the "team/integrations" label was applied to this issue.

**Comment by tjkandala** [comment]
VS Code extension states:

- Current: no automated tests or releases. Have to manually test before manually releasing.
- Good: integration test suite but manual releases. Run integration tests locally before manually releasing.
- Great: integration test suite with automatic releases. Run integration tests on release branch before automatically releasing.
  -  Nice to have: nightly builds to catch regressions relatively proactively.

**Comment by abeatrix** [comment]
@tjkandala I believe we have updated so that the test only runs nightly and is no longer in core operations, but I think we should include the test back to the main branch for PR checks since the changes made to the main branch in the past few weeks have broken the vscode build 3 times already ([1](https://github.com/sourcegraph/sourcegraph/issues/34856), [2](https://github.com/sourcegraph/sourcegraph/issues/36137), [3](https://github.com/sourcegraph/sourcegraph/pull/36649)) before the test does not run for every PR while the extension itself is using a lot of the components that are depending on the main build. 

Would love to hear your thoughts!

**Comment by philipp-spiess** [comment]
We've had issues before where changes in the Sourcegraph server was breaking the VSCE (e.g. changes in a GQL field) so I’m hesitant to only run tests on a release. Since the current state is not slowing us down, I think we can close this.
